File: Adapter/AdapterPattern.py
from abc import ABC, abstractmethod

class DesiredInterface(ABC):
    @abstractmethod
    def operation(self):
        pass

class Adapter(DesiredInterface):
    def __init__(self):
        self._adaptee = Adaptee()
        
    def operation(self):
        self._adaptee.some_operation()

class Adaptee:
    def some_operation(self):
        print("Adaptee some_request() function called.")


# Client Code
adapter = Adapter()
adapter.operation()

"""
Adaptee some_request() function called.
"""

File: Adapter/AdapterPattern.java
// Desired Interface
interface DesiredInterface {
    void operation();
}

// Adapter class
class Adapter implements DesiredInterface {
    private Adaptee adaptee;

    public Adapter() {
        this.adaptee = new Adaptee();
    }

    @Override
    public void operation() {
        adaptee.someOperation();
    }
}

// Adaptee class
class Adaptee {
    public void someOperation() {
        System.out.println("Adaptee someOperation() function called.");
    }
}

// Client Code
public class AdapterPattern {
    public static void main(String[] args) {
        DesiredInterface adapter = new Adapter();
        adapter.operation();
    }
}
